src/v/container/include/container/fragmented_vector.h

```diff
@@ -152,7 +152,7 @@ class fragmented_vector {
         --_size;
         if (_frags.back().empty()) {
             _frags.pop_back();
-            _capacity -= elems_per_frag;
+            _capacity -= std::min(elems_per_frag, _capacity);
```

Comment:
@rockwotj Is [pop_back_n](https://github.com/redpanda-data/redpanda/pull/16502/commits/aa9507b5e6cf599d53024b09c49de44f97e646f3#diff-b36ae0ae4b7e3d77bd7c38197edeee2e11613f46e232634887cad9bc3699fcf8R178) susceptible to the same issue?

Reply:
It's not because it calls `clear` if the first segment is going to be removed, otherwise the first segment will always be kept around.

Reply:
But good question :)